### ISSUE TITLE
Build fails on Android/Termux

### ISSUE THREAD

**Issue by LHLaurini** [thecoshman/http#155]
Build fails on [Termux](https://termux.dev/en/) with:
```
~/http $ cargo build
warning: file `/data/data/com.termux/files/home/http/src/main.rs` found to be present in multiple build targets:
  * `bin` target `http`
  * `bin` target `httplz`
   Compiling https v1.13.2 (/data/data/com.termux/files/home/http)
error[E0308]: mismatched types
    --> src/util/os/non_windows_non_macos.rs:39:47
     |
39   |             let ok = unsafe { ioctl(dev_file, BLKGETSIZE, &mut block_count as *mut c_ulong) } == 0;
     |                               -----           ^^^^^^^^^^ expected `i32`, found `u64`
     |                               |
     |                               arguments to this function are incorrect
     |
note: function defined here
    --> /data/data/com.termux/files/home/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.147/src/unix/linux_like/android/mod.rs:3172:12
     |
3172 |     pub fn ioctl(fd: ::c_int, request: ::c_int, ...) -> ::c_int;
     |            ^^^^^
help: you can convert a `u64` to an `i32` and panic if the converted value doesn't fit
     |
39   |             let ok = unsafe { ioctl(dev_file, BLKGETSIZE.try_into().unwrap(), &mut block_count as *mut c_ulong) } == 0;
     |                                                         ++++++++++++++++++++

For more information about this error, try `rustc --explain E0308`.
error: could not compile `https` (bin "httplz") due to previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `https` (bin "http") due to previous error
~/http $
```

It seems the signature is different for Linux and Android. While [Linux uses `::c_ulong` (`u64`)](https://github.com/rust-lang/libc/blob/835661543db1ec42a6d9a809d69c3c5b5b978b81/src/unix/linux_like/linux/gnu/mod.rs#L1260), [Android uses `::c_int` (`i32`)](https://github.com/rust-lang/libc/blob/835661543db1ec42a6d9a809d69c3c5b5b978b81/src/unix/linux_like/android/mod.rs#L3273C65-L3273C65).

The suggested fix works. I'll gladly propose a pull request if that is valid.

**Comment by nabijaczleweli** [comment]
https://lore.kernel.org/linux-man/israsi2qmpudilwpy2h6vj4dda7jofrc3oolhrcs4cpwvwzl5x@ur3oua7jxgs3/t/#u :v

**Comment by nabijaczleweli** [comment]
Wait, we even handle this:
```rust
    let ioctl_request_type = match &env::var("CARGO_CFG_TARGET_ENV").expect("CARGO_CFG_TARGET_ENV")[..] {
        "musl" => "libc::c_int",
        _ => "libc::c_ulong",
    };
```
of course, since we handle it like this instead of by autodetecting, it's bound to break when someone builds on illumos or any other non-BSD host. Cool.

**Comment by LHLaurini** [comment]
> Can you try the current `devel` branch (at least [9e6fcd1](https://github.com/thecoshman/http/commit/9e6fcd11691d01fa2e304b7a7d0fe23b859306a1))?

Sure. Now it builds correctly.

**Comment by nabijaczleweli** [comment]
Released in [`v2.0.0`](https://github.com/thecoshman/http/releases/tag/v2.0.0).
